go/stellar/stellarsvc/frontend.go

```diff
@@ -879,74 +879,95 @@ func (s *Server) MarkAsReadLocal(ctx context.Context, arg stellar1.MarkAsReadLoc
 	return nil
 }
 
-func (s *Server) IsAccountMobileOnlyLocal(ctx context.Context, arg stellar1.IsAccountMobileOnlyLocalArg) (mobileOnly bool, err error) {
+func (s *Server) IsAccountMobileOnlyLocal(ctx context.Context, arg stellar1.IsAccountMobileOnlyLocalArg) (res stellar1.AccountMobileOnly, err error) {
 	mctx, fin, err := s.Preamble(ctx, preambleArg{
 		RPCName: "IsAccountMobileOnlyLocal",
 		Err:     &err,
 	})
 	defer fin()
 	if err != nil {
-		return false, err
+		return res, err
 	}
 
 	if arg.AccountID.IsNil() {
 		mctx.Debug("IsAccountMobileOnlyLocal called with an empty account id")
-		return false, ErrAccountIDMissing
+		return res, ErrAccountIDMissing
 	}
 
-	return s.remoter.IsAccountMobileOnly(mctx.Ctx(), arg.AccountID)
+	isMobileOnly, err := s.remoter.IsAccountMobileOnly(mctx.Ctx(), arg.AccountID)
+	if err != nil {
+		return res, err
+	}
+	return stellar1.AccountMobileOnly{
+		AccountID: arg.AccountID,
+		Enabled:   isMobileOnly,
+	}, nil
 }
 
-func (s *Server) SetAccountMobileOnlyLocal(ctx context.Context, arg stellar1.SetAccountMobileOnlyLocalArg) (err error) {
+func (s *Server) SetAccountMobileOnlyLocal(ctx context.Context, arg stellar1.SetAccountMobileOnlyLocalArg) (res stellar1.AccountMobileOnly, err error) {
 	mctx, fin, err := s.Preamble(ctx, preambleArg{
 		RPCName: "SetAccountMobileOnlyLocal",
 		Err:     &err,
 	})
 	defer fin()
 	if err != nil {
-		return err
+		return res, err
 	}
 
 	if arg.AccountID.IsNil() {
 		mctx.Debug("SetAccountMobileOnlyLocal called with an empty account id")
-		return ErrAccountIDMissing
+		return res, ErrAccountIDMissing
 	}
 
 	if err = s.remoter.SetAccountMobileOnly(mctx.Ctx(), arg.AccountID); err != nil {
-		return err
+		return res, err
 	}
 
 	if err = s.walletState.UpdateAccountEntries(mctx, "set account mobile only"); err != nil {
-		return err
+		return res, err
 	}
 
-	return nil
+	isMobileOnly, err := s.remoter.IsAccountMobileOnly(mctx.Ctx(), arg.AccountID)
```

Comment:
Returning `true` would be the simplest and instant. Is this about races?

Reply:
just a stronger guarantee that the thing actually worked. i have no reason to suspect it could fail without erroring, but it costs only an http round trip to verify, which we were also doing before this PR. at least this time it doesn't include an additional roundtrip between js and go.